### PR TITLE
Separate batch of install and uninstall actions to save project system changes

### LINKED ISSUE #6035: Nuget removes analyzers from projects when updating.
Please read the following information before posting the issue.

## Details about Problem

NuGet VS UI

NuGet latest?

VS 2017

OS win10

## Detailed repro steps so we can see the same problem

1. Clone https://github.com/nestquik/diploma/commit/a7d899bc36910723a5cc1d7401f319331ea6f4ba

2. Install https://www.nuget.org/packages/PropertyChangedAnalyzers/2.0.0.1-dev

Check git diff, nothing added to projects.
Downgrade to 1.0.0 of the analyzer, works.
Upgrade to 2.0.01, removes the analyzer.

### PR DESCRIPTION
**Pull request by jainaashish**
NuGet update operation is a combination of first uninstall the existing package which also runs uninstall.ps1 and then install the new version of the same package which also runs install.ps1. We added batch start/end api during install/update operations to commit project system changes so that multiple changes to a single project are commited once and design time build is also run once. But this regressed a scenario where uninstall.ps1 powershell script makes some changes in project hierarchy which aren't commited immediately hence install.ps1 isn't able to see these changes.

This PR creates two different batches, one for all uninstall actions then commit these changes and second, for all install actions.

fixes https://github.com/NuGet/Home/issues/6035

### CODE REVIEW

File: src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs [hunk 1]
```diff
@@ -2177,34 +2177,69 @@ var projectUniqueNamesForBuildIntToUpdate
...
+                                executedNuGetProjectActions.Push(nuGetProjectAction);
+
                                 await ExecuteUninstallAsync(nuGetProject,
                                     nuGetProjectAction.PackageIdentity,
                                     packageWithDirectoriesToBeDeleted,
                                     nuGetProjectContext, token);
                             }
-                            else
+
+                            if (nuGetProjectAction.NuGetProjectActionType == NuGetProjectActionType.Uninstall)
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs]: This should be merged with the if block above, it looks like the same condition

**Review by emgarten — APPROVED** [review]
LGTM, minor comment on the double uninstall type check

File: src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs (hunk 1 above)
Comment (jainaashish) [reply src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs]: yeah, that was silly :)

**Comment by emgarten** [comment]
:shipit:

**Pull request merged**